Framework/Core/src/DispatcherDPL.cxx

```diff
@@ -48,10 +48,10 @@ void DispatcherDPL::processCallback(ProcessingContext& ctx, BernoulliGenerator&
                    << "', description '" << inputHeader->dataDescription.str
                    << "' has gSerializationMethodInvalid.";
       } else*/ if (inputHeader->payloadSerializationMethod == header::gSerializationMethodROOT) {
-        ctx.allocator().adopt(outputSpec, DataRefUtils::as<TObject>(input).release());
+        ctx.outputs().adopt(outputSpec, DataRefUtils::as<TObject>(input).release());
       } else { // POD
         // todo: use API for that when it is available
-        ctx.allocator().adoptChunk(outputSpec, const_cast<char*>(input.payload), inputHeader->size(),
+        ctx.outputs().adoptChunk(outputSpec, const_cast<char*>(input.payload), inputHeader->size(),
```

Comment:
BTW, since you are revisiting it, I remember you saying that there could be a nicer API for this particular case, especially `const_char<char *>` not being necessary.

Reply:
I will check if I can do it. I am also looking into allowing "by name" creation of outputs, so that you do not need to specify the OutputSpec in its entirety also when you use adopt / make / snapshot.

Reply:
That would help as well!